include/MeshAmalgamationBase.h

```diff
@@ -18,10 +18,12 @@ class System;
 class MeshAmalgamation {
 
 private:
+    libMesh::DofMap& freshDofMap();
 protected:
   std::string _variable_name;
   unsigned int _extra_element_integer_index;
-  unsigned int _variable_index;
+  unsigned int _synthetic_variable_index;
+  unsigned int _metric_variable_index;
```

Comment:
I'm not sure the term `synthetic` add clarity here.  Since the variable is still `_variable_name` I think it's fine to stick with `_variable_index`, and maybe use `_metric_index`...
```suggestion
  unsigned int _variable_index;
  unsigned int _metric_index;
```

Reply:
For testing, this is "synthetic" data, but in general, it's just data....